mbuild/port.py

```diff
@@ -84,3 +86,40 @@ def direction(self):
         """The unit vector pointing in the 'direction' of the Port
         """
         return unit_vector(self.xyz_with_ports[1]-self.xyz_with_ports[0])
+
+    @property
+    def access_labels(self):
+        """List of labels used to access the Port
+
+        Returns
+        -------
+        list of str
+            Strings that can be used to access this Port relative to self.root
+        """
+        access_labels = []
+        for referrer in self.referrers:
+            referrer_labels = [key for key, val in self.root.labels.items()
+                               if val == referrer]
+            port_labels = [key for key, val in referrer.labels.items()
+                           if val == self]
+            if referrer == self.root:
```

Comment:
I think you want this comparison to be done via identity: `referrer is self.root`